osvr/RenderKit/DistortionCorrectTextureCoordinate.h

```diff
@@ -0,0 +1,220 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef INCLUDED_DistortionCorrectTextureCoordinate_h_GUID_B37CBBBD_F2E1_4931_B9AC_B508C4D54F27
+#define INCLUDED_DistortionCorrectTextureCoordinate_h_GUID_B37CBBBD_F2E1_4931_B9AC_B508C4D54F27
+
+// Internal Includes
+#include "DistortionCorrectTextureCoordinate.h"
+#include "DistortionParameters.h"
+#include "UnstructuredMeshInterpolator.h"
+#include "Float2.h"
+
+// Library/third-party includes
+#include <Eigen/Core>
+
+// Standard includes
+#include <cstddef>      // for size_t
+#include <array>
+
+
+namespace osvr {
+namespace renderkit {
+
+    /// @brief Distortion-correct a texture coordinate in PresentMode
+    ///
+    /// Takes a texture coordinate that is specified in the coordinate system of
+    /// a Presented texture for a given eye, which has (0,0) at the lower left
+    /// and (1,1) at the upper right.  The lower left and upper right are at the
+    /// boundaries specified by the overfill rectangle, which are not visible
+    /// for overfill factors > 1.
+    ///
+    /// Returns the distorted location, in the same coordinate system.
+    ///
+    /// Uses the specified distortion parameters to convert to the distortion
+    /// space, distort, and convert back (also taking into account that the
+    /// distortion parameters are specified for a window that has no overfill).
+    ///
+    /// @return New coordinates on success, unchanged coordinates on failure.
+    inline Float2 DistortionCorrectTextureCoordinate(
+            size_t eye //< Eye this relates to
+            , Float2 const& inCoords //< Coordinates to modify
+            , DistortionParameters distort //< Distortion parameters
+            , size_t color //< 0 = red, 1 = green, 2 = blue
+            , float overfillFactor
+            , const std::vector<UnstructuredMeshInterpolator*>& interpolators
+            ) {
+        Float2 ret = inCoords;
+
+        // Check for invalid parameters
+        if (color > 2) {
+            return ret;
+        }
+        using Eigen::Vector2f;
+        // Convert from coordinates in the overfilled texture to coordinates
+        // that will cover the range (0,0) to (1,1) on the screen.  This is
+        // done by scaling around (0.5,0.5) to push the edges of the screen
+        // out to the (0,0) and (1,1) boundaries.
+        Eigen::Map<Vector2f> retMap(ret.data());
+        Eigen::Map<const Vector2f> inMap(inCoords.data());
+
+        Vector2f xyN = (inMap - Vector2f::Constant(0.5f)) *
+                           overfillFactor +
+                       Vector2f::Constant(0.5f);
+        /// these are for mono point that hasn't been ported to eigen yet.
```

Comment:
Here is where you'd pull out the sub-function that works on the pre-normalized components, for use by rendering engines that don't respect overfill.